### DISCUSSION TITLE
TraceQL filter on Span events

### DISCUSSION THREAD

**Discussion by ovidiubuligan** [grafana/tempo#2075] [Q&A] [answered]
Since tempo 2.0 is out now  . I was wandering if there will there be support for filtering traces that contains spans with events , and on those events filter with regex or match . https://opentelemetry.io/docs/concepts/signals/traces/#span-events 
We have a lot of valuable information in the events attached to some key spans .  Any thoughts on this ?

**Comment by joe-elliott** [comment, marked as answer]
Span events are tricky and have been explored some in an internal TraceQL spec. It's definitely on our mind and I would love to see it in the language, but it's currently competing with other features like:

https://github.com/grafana/tempo/issues?q=is%3Aissue+is%3Aopen+label%3Atraceql+

If you are able please open an issue and tag it with the `traceql` label describing what you'd like out of this. Syntax suggestions are also welcome. If you're slammed for time just let met know and I can open an issue and include some ideas as well.
